File: server/TasksManager.API/Models/Goal.cs
namespace TasksManager.API.Models;

public class Goal
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }

    public string Title { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public GoalType GoalType { get; set; }

    public DateTime? DueDate { get; set; }
    public bool IsPinned { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation
    public User User { get; set; } = null!;
    public ICollection<TaskItem> Tasks { get; set; } = new List<TaskItem>();
}

File: client/src/types/goal.ts
import { GoalType } from './enums'

export interface Goal {
  id: string

  /** User-facing title — stored as plain text; no translation key needed */
  title: string

  category: GoalCategory
  goalType: GoalType

  /** 0–100 percentage; derived from completedTasks / totalTasks */
  progress: number
  totalTasks: number
  completedTasks: number

  /** ISO date string — only relevant for GoalType.Finite */
  dueDate?: string

  /** Only for GoalType.Ongoing — tasks completed this calendar week */
  weeklyCompleted?: number
  /** Only for GoalType.Ongoing — tasks planned for this calendar week */
  weeklyTotal?: number
  /** Only for GoalType.Ongoing — tasks remaining for today */
  todayRemaining?: number

  /** Show this goal at the top of the goals list */
  isPinned?: boolean

  createdAt: string
  updatedAt: string
}
